util/progress/logs/logs.go

```diff
@@ -21,31 +30,82 @@ func newStreamWriter(ctx context.Context, stream int, printOutput bool) io.Write
 		pw:          pw,
 		stream:      stream,
 		printOutput: printOutput,
+		created:     time.Now(),
 	}
 }
 
 type streamWriter struct {
 	pw          progress.Writer
 	stream      int
 	printOutput bool
+	created     time.Time
+	size        int
+	clipping    bool
 }
 
-func (sw *streamWriter) Write(dt []byte) (int, error) {
-	sw.pw.Write(identity.NewID(), client.VertexLog{
-		Stream: sw.stream,
-		Data:   append([]byte{}, dt...),
+func (sw *streamWriter) checkLimit(n int) int {
+	configCheckOnce.Do(func() {
+		maxLogSize, err := strconv.ParseInt(os.Getenv("BUILDKIT_STEP_LOG_MAX_SIZE"), 10, 32)
```

Comment:
Shouldn't these be in config toml?

Reply:
I don't think we need to really give an impression that these should ever be set. Looks like abuse of build if you hit these (and would hit limits of ci anyway).